extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -332,14 +332,15 @@ export class AmpAnalytics extends AMP.BaseElement {
     this.config_['vars']['requestCount']++;
 
     // Replace placeholders with URI encoded values.
-    // Precedence is trigger.vars > config.vars.
+    // Precedence is trigger.vars > config.vars > event.vars.
     // Nested expansion not supported.
     request = expandTemplate(request, key => {
       const match = key.match(/([^(]*)(\([^)]*\))?/);
       const name = match[1];
       const argList = match[2] || '';
       const raw = (trigger['vars'] && trigger['vars'][name] ||
-          this.config_['vars'] && this.config_['vars'][name]);
+          this.config_['vars'] && this.config_['vars'][name]) ||
+          event.vars[name];
```

Comment:
lets say you have analytics config like following: 

```
{
  "requests": {
    "base": "https://example.com/?time=${timestamp}",
    "pageview": "${base}&name=${eventName}&type=${eventId}"
  },
  "vars": {
    "eventId": "007"
  },
  "triggers": {
    "defaultPageview": {
      "on": "visible",
      "request": "pageview",
      "vars": {
        "eventName": "page-loaded",
        "eventId": "42"
      }
    }
  }
}
```

In this case, `{     "eventId": "007"  },` defines the config var. `  "vars": {        "eventName": "page-loaded",         "eventId": "42"       }` overrides the value of eventId and this is part of trigger vars. **The question of prioritizing over event-vars is debatable**(more below). In this PR, the value comes from the trigger config (`horizontalScrollBoundary` and `verticalScrollBoundary`) but this has usefullness in other cases. Specifically #1298 and #1297 will use this to add support for `data-XYZ` attributes and `AMP_VISIBLE_*, AMP_SELECTOR_*,  AMP_HOVER_TIME, AMP_TIME_*` respectively.

About the prioritization, I am not confident what the correct prioritization should be. Having thought through this a bit more, event.vars being given highest priority is good because it is closest to the point of analytics event. On the other hand, having the priorities as they are coded right now means least amount of changes in the data that is being sent to vendors. 

Please let me know if some part is still not clear.

/cc @btownsend